lib/TableGen/Operations.cpp

```diff
@@ -447,6 +447,7 @@ bool Operation::parse(raw_ostream &errs, GenDialectsContext *context,
 
   op->m_builders.push_back(std::move(builder));
 
+  ++dialect->operationCounts[op->mnemonic];
```

Comment:
How about using a `StringSet`?
You can try inserting `op->mnemonic` into the set here. If it was already in the set (returned bool is `false`), throw an error about a duplicate definition.

Reply:
My initial idea was to just do a std::adjacent_find on the final sorted list of operations (in ::finalize), but I felt it might be more helpful if all duplicate ops are listed at once.
Since `Operation::parse` is only for a single op,I wanted to leave the actual validation work to `GenDialect`.

But honestly, I am unemotional about the final approach.

Reply:
Yeah, I although thought about that after I left the comment. Fine with me to leave it as is.